pysys/basetest.py

```diff
@@ -183,6 +185,14 @@ def cleanup(self):
 		try:
 			if self.manualTester and self.manualTester.running():
 				self.stopManualTester()
+
+			# first request them all to stop
+			# although we don't yet state this method is thread-safe, make it 
+			# as thread-safe as possible by using swap operations
+			threads, self.__backgroundThreads = list(self.__backgroundThreads), []
```

Comment:
Yeah you've persuaded me. I was trying to avoid adding lots of locking (could be a rather hgue project, and of course since 99% of tests won't be accessing these objects from multiple threads), but I'm gonna just use a lock for a few of the key data structures to add safety. I still don't want to (or have time right now) to make the classes fully thread safe (users can add their own locking as needed), but do want to reduce the chances of cleanup going wrong since the consequences of .